pkg/asset/installconfig/powervs/platform.go

```diff
@@ -0,0 +1,126 @@
+        //
+
+        longRegions := make([]string, 0, len(regions))
+	shortRegions := make([]string, 0, len(regions))
+	for id, location := range regions {
+		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
+		shortRegions = append(shortRegions, id)
+	}
+        
+	var regionTransform survey.Transformer = func(ans interface{}) interface{} {
+		switch v := ans.(type) {
+		case core.OptionAnswer:
+			return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
+		case string:
+			return strings.SplitN(v, " ", 2)[0]
+		}
+		return ""
+	}
+
+	ssn, err := GetSession()
+	if err != nil {
+	 	return nil, err
+	}
+
+	var region string
+
+	sessionRegion := ssn.Session.Region
+	if sessionRegion != "" {
+		if IsKnownRegion( sessionRegion ) {
+			region = sessionRegion
+		} else {
+			logrus.Warnf("Unrecognized powervs region %s, ignoring sessionRegion", sessionRegion)
+		}
+	}
+
+	sort.Strings(longRegions)
+	sort.Strings(shortRegions)
+	if region == "" {
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Region",
+				Help:    "The powervs region to be used for installation.",
+				// Default: fmt.Sprintf("%s (%s)", defaultRegion, regions[defaultRegion]),
+				Options: longRegions,
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				choice := regionTransform(ans).(core.OptionAnswer).Value
+				i := sort.SearchStrings(shortRegions, choice)
+				if i == len(shortRegions) || shortRegions[i] != choice {
+					return errors.Errorf("invalid region %q", choice)
+				}
+				return nil
+			}),
+			Transform: regionTransform,
+		},
+	}, &region)
+	if err != nil {
+		return nil, err
+	}}
+
+	zones := knownZones(region)
+	defaultZone := zones[0]
+
+        var zoneTransform survey.Transformer = func(ans interface{}) interface{} {
+                switch v := ans.(type) {
+                case core.OptionAnswer:
+                        return core.OptionAnswer{Value: strings.SplitN(v.Value, " ", 2)[0], Index: v.Index}
+                case string:
+                        return strings.SplitN(v, " ", 2)[0]
+                }
+                return ""
+        }
+        
+        var zone string
+        err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Zone",
+				Help:    "The powervs zone within the region to be used for installation.",
+				Default: fmt.Sprintf("%s", defaultZone),
+				Options: zones,
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				choice := zoneTransform(ans).(core.OptionAnswer).Value
+				i := sort.SearchStrings(zones, choice)
+				if i == len(zones) || zones[i] != choice {
+					return errors.Errorf("invalid zone %q", choice)
+				}
+				return nil
+			}),
+			Transform: zoneTransform,
+		},
+	}, &zone)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
we should pass through `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` to the install-config file:
```suggestion
+       if os_override := os.Getenv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); len(os_override) != 0 {
+               p.BootstrapOSImage = os_override
+               p.ClusterOSImage = os_override
+       }
+
+       p.Region = region
+       p.Zone = zone
+       p.APIKey = ssn.Creds.APIKey
+       p.UserID = ssn.Creds.UserID
+
+       return &p, nil

```